app.js

```diff
@@ -133,7 +191,27 @@ function init() {
     replaceHistoryState(APP.ConferenceUrl.getInviteUrl());
     var isUIReady = APP.UI.start();
     if (isUIReady) {
+        // Start the LogCollector's periodic "store logs" task only if we're in
+        // the conference and not on the welcome page.
+        if (APP.logCollector) {
+            APP.logCollector.start();
+            APP.logCollectorStarted = true;
+        }
+
         APP.conference.init({roomName: buildRoomName()}).then(function () {
+
+            // Will flush the logs, before the stats are disposed
+            if (APP.logCollector) {
+                APP.conference.addConferenceListener(
+                    ConferenceEvents.BEFORE_STATISTICS_DISPOSED,
```

Comment:
So the LogCollector running status does not depend directly on the statistics being started/stopped. When Jicofo sends session-terminate, because you're the only person in the room CallStats are disconnected and our wrapper is caching events and logs until they become active again (another participant joins). So while it would be ok to stop it here and start again when next participant joins it is less work this way. Maybe I should put a comment about that.